install error

Hi,

I am getting the error below when I try to install `arrayfire_r`. My system is using the most recent array fire version and cuda 6.5. Arrayfire is set up and running in c++. Any ideas?

```
-bash-4.1$ R CMD INSTALL arrayfire_r
* installing *source* package ‘arrayfire’ ...
** libs
g++ -I/usr/include/R -DNDEBUG -DAFCL -I/vega/opt/arrayfire-2014-11/include -I/usr/local/include    -fpic  -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic  -c afArray.cpp -o afArray.o
In file included from afArray.cpp:2:
R_wrapper.h:8:22: error: af/utils.h: No such file or directory
afArray.cpp:25: warning: ‘print’ initialized and declared ‘extern’
afArray.cpp:25: error: ‘SEXPREC* af::print’ redeclared as different kind of symbol
/vega/opt/arrayfire-2014-11/include/af/util.h:18: error: previous declaration of ‘void af::print(const char*, const af::array&)’
afArray.cpp:25: error: expected primary-expression before ‘A’
afArray.cpp:26: error: expected unqualified-id before ‘{’ token
make: *** [afArray.o] Error 1
ERROR: compilation failed for package ‘arrayfire’
* removing ‘...’
```

--- a/NAMESPACE
+++ b/NAMESPACE
@@ -15,8 +15,6 @@ exportMethods("Cplx")
 exportMethods("hypot")
 exportMethods("erf")
 exportMethods("erfc")
-exportMethods("erfinv")
-exportMethods("gammaln")
 
 exportMethods("mean")
 exportMethods("median")

--- a/R/afArray.R
+++ b/R/afArray.R
@@ -35,7 +35,7 @@ afArray <- function(a, type="single") {
         class(a) == "matrix") { d = dim(a) }
     else if (class(a) != "numeric") { stop("Input must be an array or numeric class") }
 
-    arr <- .Call("af_array", a, d, ty)
+    arr <- .Call("afr_array", a, d, ty)
     result <- createArray(arr)
     return (result)
 }
@@ -45,7 +45,7 @@ afArray <- function(a, type="single") {
 afRunif <- function(dims, min=0, max=1, type="single") {
     if (class(dims) != "numeric") { stop("Dimensions must be numeric") }
     dims <- as.integer(dims)
-    arr <- .Call("af_runif", dims, min, max, getTypeID(type))
+    arr <- .Call("afr_runif", dims, min, max, getTypeID(type))
     result <- createArray(arr)
     return (result)
 }
@@ -53,7 +53,7 @@ afRunif <- function(dims, min=0, max=1, type="single") {
 afRnorm <- function(dims, mean=0, sd=1, type="single") {
     if (class(dims) != "numeric") { stop("Dimensions must be numeric") }
     dims <- as.integer(dims)
-    arr <- .Call("af_rnorm", dims, mean, sd, getTypeID(type))
+    arr <- .Call("afr_rnorm", dims, mean, sd, getTypeID(type))
     result <- createArray(arr)
     return (result)
 }
@@ -61,7 +61,7 @@ afRnorm <- function(dims, mean=0, sd=1, type="single") {
 afConsts <- function(val, dims, type="single") {
     if (class(dims) != "numeric") { stop("Dimensions must be numeric") }
     dims <- as.integer(dims)
-    arr <- .Call("af_consts", val, dims, getTypeID(type))
+    arr <- .Call("afr_consts", val, dims, getTypeID(type))
     result <- createArray(arr)
     return (result)
 }
@@ -69,11 +69,11 @@ afConsts <- function(val, dims, type="single") {
 afHost <- function(a, strip=TRUE) {
     if (class(a) != "afArray") { return (a) }
     if (strip) {
-        dims <- .Call("af_dims_strip", a@ptr)
+        dims <- .Call("afr_dims_strip", a@ptr)
     } else {
-        dims <- .Call("af_dims", a@ptr)
+        dims <- .Call("afr_dims", a@ptr)
     }
-    result <- .Call("af_host", a@ptr)
+    result <- .Call("afr_host", a@ptr)
     result <- array(result, dims)
     return (result)
 }
@@ -81,21 +81,21 @@ afHost <- function(a, strip=TRUE) {
 ## Utility functions
 #####################
 afInfo <- function() {
-    invisible(.Call("af_info_r"))
+    invisible(.Call("afr_info"))
 }
 
 afPrint <- function(A) {
     if (class(A) == "afArray") {
-        invisible(.Call("af_print", A@ptr))
+        invisible(.Call("afr_print", A@ptr))
     } else {
         print(A)
     }
 }
 
 afSync <- function() {
-    invisible(.Call("af_sync_r"))
+    invisible(.Call("afr_sync"))
 }
 
 afEval <- function(A) {
-    invisible(.Call("af_eval_r", A@ptr))
+    invisible(.Call("afr_eval", A@ptr))
 }

--- a/R/afArray.R
+++ b/R/afArray.R
@@ -1,8 +1,17 @@
 setClass("afArray", representation(ptr = "externalptr"))
 
+# Hack to clean up GPU memory once in a while
+afgc <- function(flag=FALSE) {
+    memInfo <- .Call("afr_mem_info");
+    if (flag || memInfo[3] > 1000 * 1000 * 1000 || memInfo[4] > 50) {
+        invisible(gc());
+        memInfo <- .Call("afr_mem_info");
+    }
+}
+
 createArray <- function(p) {
     result <- new("afArray", ptr=p)
-    invisible(gc())
+    invisible(afgc(FALSE))
     return (result)
 }
 

--- a/R/binary.R
+++ b/R/binary.R
@@ -2,73 +2,73 @@
 ############################
 
 setMethod("+", signature(e1="afArray", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_add_aa", e1@ptr, e2@ptr)
+    ret = .Call("afr_add_aa", e1@ptr, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("-", signature(e1="afArray", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_sub_aa", e1@ptr, e2@ptr)
+    ret = .Call("afr_sub_aa", e1@ptr, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("*", signature(e1="afArray", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_mul_aa", e1@ptr, e2@ptr)
+    ret = .Call("afr_mul_aa", e1@ptr, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("/", signature(e1="afArray", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_div_aa", e1@ptr, e2@ptr)
+    ret = .Call("afr_div_aa", e1@ptr, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("<", signature(e1="afArray", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_lt_aa", e1@ptr, e2@ptr)
+    ret = .Call("afr_lt_aa", e1@ptr, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("<=", signature(e1="afArray", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_le_aa", e1@ptr, e2@ptr)
+    ret = .Call("afr_le_aa", e1@ptr, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod(">", signature(e1="afArray", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_gt_aa", e1@ptr, e2@ptr)
+    ret = .Call("afr_gt_aa", e1@ptr, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod(">=", signature(e1="afArray", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_ge_aa", e1@ptr, e2@ptr)
+    ret = .Call("afr_ge_aa", e1@ptr, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("==", signature(e1="afArray", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_eq_aa", e1@ptr, e2@ptr)
+    ret = .Call("afr_eq_aa", e1@ptr, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("!=", signature(e1="afArray", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_ne_aa", e1@ptr, e2@ptr)
+    ret = .Call("afr_ne_aa", e1@ptr, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("^", signature(e1="afArray", e2="afArray"),  function(e1, e2) {
-    ret = .Call("af_pow_aa", e1@ptr, e2@ptr)
+    ret = .Call("afr_pow_aa", e1@ptr, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("%%", signature(e1="afArray", e2="afArray"),  function(e1, e2) {
-    ret = .Call("af_mod_aa", e1@ptr, e2@ptr)
+    ret = .Call("afr_mod_aa", e1@ptr, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
@@ -77,73 +77,73 @@ setMethod("%%", signature(e1="afArray", e2="afArray"),  function(e1, e2) {
 ##############################
 
 setMethod("+", signature(e1="afArray", e2="numeric"), function(e1, e2) {
-    ret = .Call("af_add_an", e1@ptr, e2)
+    ret = .Call("afr_add_an", e1@ptr, e2)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("-", signature(e1="afArray", e2="numeric"), function(e1, e2) {
-    ret = .Call("af_sub_an", e1@ptr, e2)
+    ret = .Call("afr_sub_an", e1@ptr, e2)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("*", signature(e1="afArray", e2="numeric"), function(e1, e2) {
-    ret = .Call("af_mul_an", e1@ptr, e2)
+    ret = .Call("afr_mul_an", e1@ptr, e2)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("/", signature(e1="afArray", e2="numeric"), function(e1, e2) {
-    ret = .Call("af_div_an", e1@ptr, e2)
+    ret = .Call("afr_div_an", e1@ptr, e2)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("<", signature(e1="afArray", e2="numeric"), function(e1, e2) {
-    ret = .Call("af_lt_an", e1@ptr, e2)
+    ret = .Call("afr_lt_an", e1@ptr, e2)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("<=", signature(e1="afArray", e2="numeric"), function(e1, e2) {
-    ret = .Call("af_le_an", e1@ptr, e2)
+    ret = .Call("afr_le_an", e1@ptr, e2)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod(">", signature(e1="afArray", e2="numeric"), function(e1, e2) {
-    ret = .Call("af_gt_an", e1@ptr, e2)
+    ret = .Call("afr_gt_an", e1@ptr, e2)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod(">=", signature(e1="afArray", e2="numeric"), function(e1, e2) {
-    ret = .Call("af_ge_an", e1@ptr, e2)
+    ret = .Call("afr_ge_an", e1@ptr, e2)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("==", signature(e1="afArray", e2="numeric"), function(e1, e2) {
-    ret = .Call("af_eq_an", e1@ptr, e2)
+    ret = .Call("afr_eq_an", e1@ptr, e2)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("!=", signature(e1="afArray", e2="numeric"), function(e1, e2) {
-    ret = .Call("af_ne_an", e1@ptr, e2)
+    ret = .Call("afr_ne_an", e1@ptr, e2)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("^", signature(e1="afArray", e2="numeric"), function(e1, e2) {
-    ret = .Call("af_pow_an", e1@ptr, e2)
+    ret = .Call("afr_pow_an", e1@ptr, e2)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("%%", signature(e1="afArray", e2="numeric"), function(e1, e2) {
-    ret = .Call("af_mod_an", e1@ptr, e2)
+    ret = .Call("afr_mod_an", e1@ptr, e2)
     result <- createArray(ret)
     return (result)
 })
@@ -152,73 +152,73 @@ setMethod("%%", signature(e1="afArray", e2="numeric"), function(e1, e2) {
 ##############################
 
 setMethod("+", signature(e1="numeric", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_add_na", e1, e2@ptr)
+    ret = .Call("afr_add_na", e1, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("-", signature(e1="numeric", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_sub_na", e1, e2@ptr)
+    ret = .Call("afr_sub_na", e1, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("*", signature(e1="numeric", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_mul_na", e1, e2@ptr)
+    ret = .Call("afr_mul_na", e1, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("/", signature(e1="numeric", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_div_na", e1, e2@ptr)
+    ret = .Call("afr_div_na", e1, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("<", signature(e1="numeric", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_lt_na", e1, e2@ptr)
+    ret = .Call("afr_lt_na", e1, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("<=", signature(e1="numeric", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_le_na", e1, e2@ptr)
+    ret = .Call("afr_le_na", e1, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod(">", signature(e1="numeric", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_gt_na", e1, e2@ptr)
+    ret = .Call("afr_gt_na", e1, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod(">=", signature(e1="numeric", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_ge_na", e1, e2@ptr)
+    ret = .Call("afr_ge_na", e1, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("==", signature(e1="numeric", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_eq_na", e1, e2@ptr)
+    ret = .Call("afr_eq_na", e1, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("!=", signature(e1="numeric", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_ne_na", e1, e2@ptr)
+    ret = .Call("afr_ne_na", e1, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("^", signature(e1="numeric", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_pow_na", e1, e2@ptr)
+    ret = .Call("afr_pow_na", e1, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
 
 setMethod("%%", signature(e1="numeric", e2="afArray"), function(e1, e2) {
-    ret = .Call("af_mod_na", e1, e2@ptr)
+    ret = .Call("afr_mod_na", e1, e2@ptr)
     result <- createArray(ret)
     return (result)
 })
@@ -230,63 +230,63 @@ setGeneric("hypot", function(x, y) standardGeneric("hypot"))
 
 setMethod("Cplx", signature("afArray", "afArray"),
           function(real, imag) {
-              ret = .Call("af_complex_aa", real@ptr, imag@ptr)
+              ret = .Call("afr_complex_aa", real@ptr, imag@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("Cplx", signature("afArray", "numeric"),
           function(real, imag=0) {
-              ret = .Call("af_complex_an", real@ptr, imag)
+              ret = .Call("afr_complex_an", real@ptr, imag)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("Cplx", signature("numeric", "afArray"),
           function(real=0, imag) {
-              ret = .Call("af_complex_na", real, imag@ptr)
+              ret = .Call("afr_complex_na", real, imag@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("atan2", signature(y="afArray", x="afArray"),
           function(y, x) {
-              ret = .Call("af_atan2_aa", y@ptr, x@ptr)
+              ret = .Call("afr_atan2_aa", y@ptr, x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("atan2", signature(y="afArray", x="numeric"),
           function(y, x) {
-              ret = .Call("af_atan2_an", y@ptr, x)
+              ret = .Call("afr_atan2_an", y@ptr, x)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("atan2", signature(y="numeric", x="afArray"),
           function(y, x) {
-              ret = .Call("af_atan2_na", x, y@ptr)
+              ret = .Call("afr_atan2_na", x, y@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("hypot", signature("afArray", "afArray"),
           function(x, y) {
-              ret = .Call("af_hypot_aa", x@ptr, y@ptr)
+              ret = .Call("afr_hypot_aa", x@ptr, y@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("hypot", signature("afArray", "numeric"),
           function(x, y) {
-              ret = .Call("af_hypot_an", x@ptr, y)
+              ret = .Call("afr_hypot_an", x@ptr, y)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("hypot", signature("numeric", "afArray"),
           function(x, y) {
-              ret = .Call("af_hypot_na", x, y@ptr)
+              ret = .Call("afr_hypot_na", x, y@ptr)
               result <- createArray(ret)
               return (result)
           })

--- a/R/classify.R
+++ b/R/classify.R
@@ -2,7 +2,7 @@ setGeneric("afKNN1", function(train, test, cl) standardGeneric("afKNN1"))
 
 setMethod("afKNN1", signature(train="afArray", test="afArray", cl="afArray"),
           function(train, test, cl) {
-              ret = .Call("af_knn1", train@ptr, test@ptr, cl@ptr)
+              ret = .Call("afr_knn1", train@ptr, test@ptr, cl@ptr)
               result <- createArray(ret)
               return(result)
           })

--- a/R/core.R
+++ b/R/core.R
@@ -3,35 +3,35 @@
 
 setMethod("sum", signature(x="afArray", na.rm="logical"),
           function(x, na.rm=0) {
-              ret = .Call("af_sum", x@ptr, -1)
+              ret = .Call("afr_sum", x@ptr, -1)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("any", signature(x="afArray", na.rm="logical"),
           function(x, na.rm=0) {
-              ret = .Call("af_anytrue", x@ptr, -1)
+              ret = .Call("afr_anytrue", x@ptr, -1)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("all", signature(x="afArray", na.rm="logical"),
           function(x, na.rm=0) {
-              ret = .Call("af_alltrue", x@ptr, -1)
+              ret = .Call("afr_alltrue", x@ptr, -1)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("min", signature(x="afArray", na.rm="logical"),
           function(x, na.rm=0) {
-              ret = .Call("af_min", x@ptr, -1)
+              ret = .Call("afr_min", x@ptr, -1)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("max", signature(x="afArray", na.rm="logical"),
           function(x, na.rm=0) {
-              ret = .Call("af_max", x@ptr, -1)
+              ret = .Call("afr_max", x@ptr, -1)
               result <- createArray(ret)
               return (result)
           })
@@ -39,7 +39,7 @@ setMethod("max", signature(x="afArray", na.rm="logical"),
 setGeneric("mean", function(x) standardGeneric("mean"))
 setMethod("mean", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_mean", x@ptr, -1)
+              ret = .Call("afr_mean", x@ptr, -1)
               result <- createArray(ret)
               return (result)
           })
@@ -47,7 +47,7 @@ setMethod("mean", signature(x="afArray"),
 setGeneric("median", function(x) standardGeneric("median"))
 setMethod("median", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_median", x@ptr, -1)
+              ret = .Call("afr_median", x@ptr, -1)
               result <- createArray(ret)
               return (result)
           })
@@ -64,49 +64,49 @@ setGeneric("dimMedian", function(x, dim) standardGeneric("dimMedian"))
 
 setMethod("dimSum", signature(x="afArray", dim="numeric"),
           function(x, dim=1) {
-              ret = .Call("af_sum", x@ptr, dim-1)
+              ret = .Call("afr_sum", x@ptr, dim-1)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("dimMin", signature(x="afArray", dim="numeric"),
           function(x, dim=1) {
-              ret = .Call("af_min", x@ptr, dim-1)
+              ret = .Call("afr_min", x@ptr, dim-1)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("dimMax", signature(x="afArray", dim="numeric"),
           function(x, dim=1) {
-              ret = .Call("af_max", x@ptr, dim-1)
+              ret = .Call("afr_max", x@ptr, dim-1)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("dimAny", signature(x="afArray", dim="numeric"),
           function(x, dim=1) {
-              ret = .Call("af_any", x@ptr, dim-1)
+              ret = .Call("afr_any", x@ptr, dim-1)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("dimAll", signature(x="afArray", dim="numeric"),
           function(x, dim=1) {
-              ret = .Call("af_all", x@ptr, dim-1)
+              ret = .Call("afr_all", x@ptr, dim-1)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("dimMean", signature(x="afArray", dim="numeric"),
           function(x, dim=1) {
-              ret = .Call("af_mean", x@ptr, dim-1)
+              ret = .Call("afr_mean", x@ptr, dim-1)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("dimMedian", signature(x="afArray", dim="numeric"),
           function(x, dim=1) {
-              ret = .Call("af_median", x@ptr, dim-1)
+              ret = .Call("afr_median", x@ptr, dim-1)
               result <- createArray(ret)
               return (result)
           })

--- a/R/index.R
+++ b/R/index.R
@@ -14,13 +14,13 @@ setMethod("[", signature(x="afArray", i="afArray", j="afArray"),
               indices <- getIndices(...)
 
               if (missing(i)) {
-                  ret =.Call("af_get_sass", x@ptr, 0, j@ptr, 0, 0)
+                  ret =.Call("afr_get_sass", x@ptr, 0, j@ptr, 0, 0)
               }
 
               if (missing(j)) {
-                  ret =.Call("af_get_asss", x@ptr, i@ptr, 0, 0, 0)
+                  ret =.Call("afr_get_asss", x@ptr, i@ptr, 0, 0, 0)
               }
-              ret = .Call("af_get_aass", x@ptr, i@ptr, j@ptr, 0, 0)
+              ret = .Call("afr_get_aass", x@ptr, i@ptr, j@ptr, 0, 0)
               result = createArray(ret)
               return (result)
           })
@@ -30,16 +30,16 @@ setMethod("[", signature(x="afArray", i="numeric", j="afArray"),
               indices <- getIndices(...)
 
               if (missing(i)) {
-                  ret =.Call("af_get_sass", x@ptr, 0, j@ptr, 0, 0)
+                  ret =.Call("afr_get_sass", x@ptr, 0, j@ptr, 0, 0)
               } else {
                   i = as.integer(i)
               }
 
               if (missing(j)) {
-                  ret =.Call("af_get_nsss", x@ptr, i, 0, 0, 0)
+                  ret =.Call("afr_get_nsss", x@ptr, i, 0, 0, 0)
               }
 
-              ret = .Call("af_get_nass", x@ptr, i, j@ptr, 0, 0)
+              ret = .Call("afr_get_nass", x@ptr, i, j@ptr, 0, 0)
               result = createArray(ret)
               return (result)
           })
@@ -49,16 +49,16 @@ setMethod("[", signature(x="afArray", i="afArray", j="numeric"),
               indices <- getIndices(...)
 
               if (missing(j)) {
-                  ret =.Call("af_get_asss", x@ptr, i@ptr, 0, 0, 0)
+                  ret =.Call("afr_get_asss", x@ptr, i@ptr, 0, 0, 0)
               } else {
                   j = as.integer(j)
               }
 
               if (missing(i)) {
-                  ret =.Call("af_get_snss", x@ptr, 0, j, 0, 0)
+                  ret =.Call("afr_get_snss", x@ptr, 0, j, 0, 0)
               }
 
-              ret = .Call("af_get_anss", x@ptr, i@ptr, j, 0, 0)
+              ret = .Call("afr_get_anss", x@ptr, i@ptr, j, 0, 0)
               result = createArray(ret)
               return (result)
           })
@@ -73,9 +73,9 @@ setMethod("[", signature(x="afArray", i="ANY", j="ANY"),
               j = as.integer(j)
 
               if (sum(i) == 0) {
-                  ret = .Call("af_get_snss", x@ptr, 0, j, 0, 0)
+                  ret = .Call("afr_get_snss", x@ptr, 0, j, 0, 0)
               } else {
-                  ret = .Call("af_get_nnss", x@ptr, i, j, 0, 0)
+                  ret = .Call("afr_get_nnss", x@ptr, i, j, 0, 0)
               }
 
               result = createArray(ret)
@@ -88,14 +88,14 @@ setMethod("[<-", signature(x="afArray", i="afArray", j="afArray"),
               if (class(value) != "afArray") { stop ("value parameter must be afArray") }
 
               if (missing(i)) {
-                  ret =.Call("af_set_sass", x@ptr, value@ptr, 0, j@ptr, 0, 0)
+                  ret =.Call("afr_set_sass", x@ptr, value@ptr, 0, j@ptr, 0, 0)
               }
 
               if (missing(j)) {
-                  ret =.Call("af_set_asss", x@ptr, value@ptr, i@pr, 0, 0, 0)
+                  ret =.Call("afr_set_asss", x@ptr, value@ptr, i@pr, 0, 0, 0)
               }
 
-              ret = .Call("af_set_aass", x@ptr, value@ptr, i@ptr, j@ptr, 0, 0)
+              ret = .Call("afr_set_aass", x@ptr, value@ptr, i@ptr, j@ptr, 0, 0)
               result = createArray(ret)
               return (result)
           })
@@ -106,16 +106,16 @@ setMethod("[<-", signature(x="afArray", i="numeric", j="afArray"),
               if (class(value) != "afArray") { stop ("value parameter must be afArray") }
 
               if (missing(i)) {
-                  ret =.Call("af_set_sass", x@ptr, value@ptr, 0, j@ptr, 0, 0)
+                  ret =.Call("afr_set_sass", x@ptr, value@ptr, 0, j@ptr, 0, 0)
               } else {
                   i = as.integer(i)
               }
 
               if (missing(j)) {
-                  ret =.Call("af_set_nsss", x@ptr, value@ptr, i, 0, 0, 0)
+                  ret =.Call("afr_set_nsss", x@ptr, value@ptr, i, 0, 0, 0)
               }
 
-              ret = .Call("af_set_nass", x@ptr, value@ptr, i, j@ptr, 0, 0)
+              ret = .Call("afr_set_nass", x@ptr, value@ptr, i, j@ptr, 0, 0)
               result = createArray(ret)
               return (result)
           })
@@ -126,16 +126,16 @@ setMethod("[<-", signature(x="afArray", i="afArray", j="numeric"),
               if (class(value) != "afArray") { stop ("value parameter must be afArray") }
 
               if (missing(j)) {
-                  ret =.Call("af_set_asss", x@ptr, value@ptr, i@ptr, 0, 0, 0)
+                  ret =.Call("afr_set_asss", x@ptr, value@ptr, i@ptr, 0, 0, 0)
               } else {
                   j = as.integer(j)
               }
 
               if (missing(i)) {
-                  ret =.Call("af_set_snss", x@ptr, value@ptr, 0, j, 0, 0)
+                  ret =.Call("afr_set_snss", x@ptr, value@ptr, 0, j, 0, 0)
               }
 
-              ret = .Call("af_set_anss", x@ptr, value@ptr, i@ptr, j, 0, 0)
+              ret = .Call("afr_set_anss", x@ptr, value@ptr, i@ptr, j, 0, 0)
               result = createArray(ret)
               return (result)
           })
@@ -151,9 +151,9 @@ setMethod("[<-", signature(x="afArray", i="ANY", j="ANY"),
               j = as.integer(j)
 
               if (sum(i) == 0) {
-                  ret = .Call("af_set_snss", x@ptr, value@ptr, 0, j, 0, 0)
+                  ret = .Call("afr_set_snss", x@ptr, value@ptr, 0, j, 0, 0)
               } else {
-                  ret = .Call("af_set_nnss", x@ptr, value@ptr, i, j, 0, 0)
+                  ret = .Call("afr_set_nnss", x@ptr, value@ptr, i, j, 0, 0)
               }
 
               result = createArray(ret)

--- a/R/stats.R
+++ b/R/stats.R
@@ -1,7 +1,15 @@
 setGeneric("sd", function(x) standardGeneric("sd"))
 setMethod("sd", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_stdev", x@ptr)
+              ret = .Call("afr_stdev", x@ptr)
+              result <- createArray(ret)
+              return (result)
+          })
+
+setGeneric("var", function(x) standardGeneric("var"))
+setMethod("var", signature(x="afArray"),
+          function(x) {
+              ret = .Call("afr_var", x@ptr)
               result <- createArray(ret)
               return (result)
           })
@@ -10,24 +18,10 @@ setGeneric("cov", function(x,y) standardGeneric("cov"))
 setMethod("cov", signature(x="afArray", y="ANY"),
           function(x, y=NULL) {
               if (y == NULL) {
-                  ret = .Call("af_cov", x@ptr)
+                  ret = .Call("afr_var", x@ptr)
               } else {
-                  ret = .Call("af_cov2", x@ptr, y@ptr)
+                  ret = .Call("afr_cov2", x@ptr, y@ptr)
               }
               result <- createArray(ret)
               return (result)
           })
-
-setGeneric("var", function(x, y) standardGeneric("var"))
-setMethod("var", signature(x="afArray", y="ANY"),
-          function(x, y=NULL) {
-              if (y == NULL) {
-                  ret = .Call("af_var", x@ptr)
-              } else {
-                  ret = .Call("af_var2", x@ptr, y@ptr)
-              }
-              result <- createArray(ret)
-              return (result)
-          })
-
-

--- a/R/unary.R
+++ b/R/unary.R
@@ -3,42 +3,42 @@
 
 setMethod("sin", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_sin", x@ptr)
+              ret = .Call("afr_sin", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("cos", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_cos", x@ptr)
+              ret = .Call("afr_cos", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("tan", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_tan", x@ptr)
+              ret = .Call("afr_tan", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("asin", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_asin", x@ptr)
+              ret = .Call("afr_asin", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("acos", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_acos", x@ptr)
+              ret = .Call("afr_acos", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("atan", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_atan", x@ptr)
+              ret = .Call("afr_atan", x@ptr)
               result <- createArray(ret)
               return (result)
           })
@@ -48,42 +48,42 @@ setMethod("atan", signature(x="afArray"),
 
 setMethod("sinh", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_sinh", x@ptr)
+              ret = .Call("afr_sinh", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("cosh", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_cosh", x@ptr)
+              ret = .Call("afr_cosh", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("tanh", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_tanh", x@ptr)
+              ret = .Call("afr_tanh", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("asinh", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_asinh", x@ptr)
+              ret = .Call("afr_asinh", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("acosh", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_acosh", x@ptr)
+              ret = .Call("afr_acosh", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("atanh", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_atanh", x@ptr)
+              ret = .Call("afr_atanh", x@ptr)
               result <- createArray(ret)
               return (result)
           })
@@ -93,28 +93,28 @@ setMethod("atanh", signature(x="afArray"),
 
 setMethod("log", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_log", x@ptr)
+              ret = .Call("afr_log", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
-setMethod("log2", signature(x="afArray"),
-          function(x) {
-              ret = .Call("af_log2", x@ptr)
-              result <- createArray(ret)
-              return (result)
-          })
+## setMethod("log2", signature(x="afArray"),
+##           function(x) {
+##               ret = .Call("afr_log2", x@ptr)
+##               result <- createArray(ret)
+##               return (result)
+##           })
 
 setMethod("log10", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_log10", x@ptr)
+              ret = .Call("afr_log10", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("log1p", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_log1p", x@ptr)
+              ret = .Call("afr_log1p", x@ptr)
               result <- createArray(ret)
               return (result)
           })
@@ -123,54 +123,54 @@ setMethod("log1p", signature(x="afArray"),
 ##########################
 setGeneric("erf", function(x) standardGeneric("erf"))
 setGeneric("erfc", function(x) standardGeneric("erfc"))
-setGeneric("erfinv", function(x) standardGeneric("erfinv"))
-setGeneric("gammaln", function(x) standardGeneric("gammaln"))
+#setGeneric("erfinv", function(x) standardGeneric("erfinv"))
+#setGeneric("lgamma", function(x) standardGeneric("lgamma"))
 
 setMethod("exp", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_exp", x@ptr)
+              ret = .Call("afr_exp", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("expm1", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_expm1", x@ptr)
+              ret = .Call("afr_expm1", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("erf", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_erf", x@ptr)
+              ret = .Call("afr_erf", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("erfc", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_erfc", x@ptr)
+              ret = .Call("afr_erfc", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
-setMethod("erfinv", signature(x="afArray"),
-          function(x) {
-              ret = .Call("af_erfinv", x@ptr)
-              result <- createArray(ret)
-              return (result)
-          })
+## setMethod("erfinv", signature(x="afArray"),
+##           function(x) {
+##               ret = .Call("afr_erfinv", x@ptr)
+##               result <- createArray(ret)
+##               return (result)
+##           })
 
 setMethod("gamma", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_gamma", x@ptr)
+              ret = .Call("afr_tgamma", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
-setMethod("gammaln", signature(x="afArray"),
+setMethod("lgamma", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_gammaln", x@ptr)
+              ret = .Call("afr_lgamma", x@ptr)
               result <- createArray(ret)
               return (result)
           })
@@ -180,73 +180,73 @@ setMethod("gammaln", signature(x="afArray"),
 
 setMethod("sqrt", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_sqrt", x@ptr)
+              ret = .Call("afr_sqrt", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("abs", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_abs", x@ptr)
+              ret = .Call("afr_abs", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("round", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_round", x@ptr)
+              ret = .Call("afr_round", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("ceiling", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_ceil", x@ptr)
+              ret = .Call("afr_ceil", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("floor", signature(x="afArray"),
           function(x) {
-              ret = .Call("af_floor", x@ptr)
+              ret = .Call("afr_floor", x@ptr)
               result <- createArray(ret)
               return (result)
           })
 
-setMethod("trunc", signature(x="afArray"),
-          function(x) {
-              ret = .Call("af_trunc", x@ptr)
-              result <- createArray(ret)
-              return (result)
-          })
+## setMethod("trunc", signature(x="afArray"),
+##           function(x) {
+##               ret = .Call("afr_trunc", x@ptr)
+##               result <- createArray(ret)
+##               return (result)
+##           })
 
-setMethod("sign", signature(x="afArray"),
-          function(x) {
-              ret = .Call("af_sign", x@ptr)
-              result <- createArray(ret)
-              return (result)
-          })
+## setMethod("sign", signature(x="afArray"),
+##           function(x) {
+##               ret = .Call("afr_sign", x@ptr)
+##               result <- createArray(ret)
+##               return (result)
+##           })
 
 ## Complex functions
 ####################
 
 setMethod("Re", signature(z="afArray"),
           function(z) {
-              ret = .Call("af_real", z@ptr)
+              ret = .Call("afr_real", z@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("Im", signature(z="afArray"),
           function(z) {
-              ret = .Call("af_imag", z@ptr)
+              ret = .Call("afr_imag", z@ptr)
               result <- createArray(ret)
               return (result)
           })
 
 setMethod("Conj", signature(z="afArray"),
           function(z) {
-              ret = .Call("af_conjg", z@ptr)
+              ret = .Call("afr_conjg", z@ptr)
               result <- createArray(ret)
               return (result)
           })

--- a/README.md
+++ b/README.md
@@ -6,7 +6,7 @@ This repository contains the files required to use ArrayFire from R.
 Prerequisites
 ---------------
 
-- The latest version of ArrayFire. You can [download here](http://www.arrayfire.com/docs/installation.htm)
+- Clone and build [ArrayFire](https://github.com/arrayfire/arrayfire)
     - All the pre-requisites for ArrayFire still apply.
 
 - **make**
@@ -26,7 +26,7 @@ Usage
 
 After you the necessary pre-requisites, edit the following paramets
 
-- Open **src/Makevars** and change **AF_PATH** and **CUDA_PATH** to the right location
+- Open **src/Makevars** and change **AF_PATH** to the right location
 - Change **AFCL** to 1 if you are using OpenCL
 
 ### Linux

--- a/src/Makevars
+++ b/src/Makevars
@@ -1,21 +1,14 @@
 # Changepath
-AF_PATH?=/opt/arrayfire-2.0
+AF_PATH?=/usr/local
 AFCL?=0
-CUDA_PATH?=/usr/local/cuda
 
 ifeq ($(AFCL), 0)
-     AF_LIB=afcu
-     PKG_CPPFLAGS+=-I$(CUDA_PATH)/include
+     AF_LIB=afcuda
 else
-     AF_LIB=afcl
-     PKG_CPPFLAGS+=-DAFCL
+     AF_LIB=afopencl
 endif
 
-ifeq ($(shell uname -m), x86_64)
-  LIB:=lib64
-else
-  LIB:=lib
-endif
+LIB=lib
 
 PKG_CFLAGS=
 PKG_CPPFLAGS+=-I$(AF_PATH)/include

--- a/src/R_wrapper.cpp
+++ b/src/R_wrapper.cpp
@@ -2,7 +2,7 @@
 #include "R_wrapper.h"
 using namespace af;
 
-static void af_destroy(SEXP A)
+static void afr_destroy(SEXP A)
 {
     try {
         array *a = getPtr(A);
@@ -15,7 +15,7 @@ static void af_destroy(SEXP A)
 SEXP getSEXP(array *ptr)
 {
     SEXP res = R_MakeExternalPtr(ptr, R_NilValue, R_NilValue);
-    R_RegisterCFinalizerEx(res, af_destroy, TRUE);
+    R_RegisterCFinalizerEx(res, afr_destroy, TRUE);
     return res;
 }
 

--- a/src/R_wrapper.h
+++ b/src/R_wrapper.h
@@ -5,7 +5,7 @@
 #include <Rdefines.h>
 #undef eval // defined in Rinternals.h
 #include <arrayfire.h>
-#include <af/utils.h>
+#include <af/util.h>
 
 #ifdef __cplusplus
 #define EXTERNC extern "C" SEXP

--- a/src/afArray.cpp
+++ b/src/afArray.cpp
@@ -2,6 +2,7 @@
 #include "R_wrapper.h"
 using namespace af;
 
+
 EXTERNC afr_info()
 {
     try {
@@ -21,6 +22,34 @@ EXTERNC afr_sync()
         error_return(ae.what());
     }
 }
+
+EXTERNC afr_mem_info()
+{
+    try {
+        SEXP result = PROTECT(allocVector(REALSXP, 4));
+        size_t alloc_bytes;
+        size_t alloc_buffers;
+        size_t lock_bytes;
+        size_t lock_buffers;
+
+        deviceMemInfo(&alloc_bytes,
+                      &alloc_buffers,
+                      &lock_bytes,
+                      &lock_buffers);
+
+        REAL(result)[0] = (double)alloc_bytes;
+        REAL(result)[1] = (double)alloc_buffers;
+        REAL(result)[2] = (double)lock_bytes;
+        REAL(result)[3] = (double)lock_buffers;
+
+        UNPROTECT(1);
+
+        return result;
+    } catch (af::exception &ae) {
+        error_return(ae.what());
+    }
+}
+
 
 EXTERNC afr_print(SEXP A)
 {

--- a/src/afArray.cpp
+++ b/src/afArray.cpp
@@ -2,7 +2,7 @@
 #include "R_wrapper.h"
 using namespace af;
 
-EXTERNC af_info_r()
+EXTERNC afr_info()
 {
     try {
         af::info();
@@ -12,7 +12,7 @@ EXTERNC af_info_r()
     }
 }
 
-EXTERNC af_sync_r()
+EXTERNC afr_sync()
 {
     try {
         af::sync();
@@ -22,18 +22,18 @@ EXTERNC af_sync_r()
     }
 }
 
-EXTERNC af_print(SEXP A)
+EXTERNC afr_print(SEXP A)
 {
     try {
         array *a = getPtr(A);
-        _print("afArray", *a);
+        af::print("afArray", *a);
         return (R_NilValue);
     } catch (af::exception &ae) {
         error_return(ae.what());
     }
 }
 
-EXTERNC af_dims(SEXP A)
+EXTERNC afr_dims(SEXP A)
 {
     try {
         array *a = getPtr(A);
@@ -46,7 +46,7 @@ EXTERNC af_dims(SEXP A)
     }
 }
 
-EXTERNC af_dims_strip(SEXP A)
+EXTERNC afr_dims_strip(SEXP A)
 {
     try {
         array *a = getPtr(A);
@@ -70,7 +70,7 @@ EXTERNC af_dims_strip(SEXP A)
     }
 }
 
-EXTERNC af_host(SEXP A)
+EXTERNC afr_host(SEXP A)
 {
     try {
         array *a = getPtr(A);
@@ -104,7 +104,7 @@ EXTERNC af_host(SEXP A)
     }
 }
 
-EXTERNC af_array(SEXP data, SEXP _dims, SEXP _type)
+EXTERNC afr_array(SEXP data, SEXP _dims, SEXP _type)
 {
     try {
         dtype ty = (dtype)*(IntPtr(_type, 0));
@@ -129,7 +129,7 @@ EXTERNC af_array(SEXP data, SEXP _dims, SEXP _type)
     }
 }
 
-EXTERNC af_runif(SEXP _dims, SEXP _mn, SEXP _mx, SEXP _type)
+EXTERNC afr_runif(SEXP _dims, SEXP _mn, SEXP _mx, SEXP _type)
 {
     try {
         dtype ty = (dtype)*(IntPtr(_type, 0));
@@ -145,7 +145,7 @@ EXTERNC af_runif(SEXP _dims, SEXP _mn, SEXP _mx, SEXP _type)
     }
 }
 
-EXTERNC af_rnorm(SEXP _dims, SEXP _mean, SEXP _sd, SEXP _type)
+EXTERNC afr_rnorm(SEXP _dims, SEXP _mean, SEXP _sd, SEXP _type)
 {
     try {
         dtype ty = (dtype)*(IntPtr(_type, 0));
@@ -161,7 +161,7 @@ EXTERNC af_rnorm(SEXP _dims, SEXP _mean, SEXP _sd, SEXP _type)
     }
 }
 
-EXTERNC af_consts(SEXP _val, SEXP _dims,  SEXP _type)
+EXTERNC afr_consts(SEXP _val, SEXP _dims,  SEXP _type)
 {
     try {
         dtype ty = (dtype)*(IntPtr(_type, 0));

--- a/src/binary.cpp
+++ b/src/binary.cpp
@@ -3,7 +3,7 @@
 using namespace af;
 
 #define BINARY(fn, op)                          \
-    EXTERNC af_##fn##_aa(SEXP A, SEXP B)        \
+    EXTERNC afr_##fn##_aa(SEXP A, SEXP B)        \
     {                                           \
         try {                                   \
             array *a = getPtr(A);               \
@@ -16,7 +16,7 @@ using namespace af;
         }                                       \
     }                                           \
                                                 \
-    EXTERNC af_##fn##_an(SEXP A, SEXP B)        \
+    EXTERNC afr_##fn##_an(SEXP A, SEXP B)        \
     {                                           \
         try{                                    \
             array *a = getPtr(A);               \
@@ -29,7 +29,7 @@ using namespace af;
         }                                       \
     }                                           \
                                                 \
-    EXTERNC af_##fn##_na(SEXP A, SEXP B)        \
+    EXTERNC afr_##fn##_na(SEXP A, SEXP B)        \
     {                                           \
         try{                                    \
             double a = *RealPtr(A, 0);          \
@@ -56,7 +56,7 @@ BINARY(ne ,!=)
 #undef BINARY
 
 #define BINARY(fn)                              \
-    EXTERNC af_##fn##_aa(SEXP A, SEXP B)        \
+    EXTERNC afr_##fn##_aa(SEXP A, SEXP B)        \
     {                                           \
         try{                                    \
             array *a = getPtr(A);               \
@@ -69,7 +69,7 @@ BINARY(ne ,!=)
         }                                       \
     }                                           \
                                                 \
-    EXTERNC af_##fn##_an(SEXP A, SEXP B)        \
+    EXTERNC afr_##fn##_an(SEXP A, SEXP B)        \
     {                                           \
         try{                                    \
             array *a = getPtr(A);               \
@@ -82,7 +82,7 @@ BINARY(ne ,!=)
         }                                       \
     }                                           \
                                                 \
-    EXTERNC af_##fn##_na(SEXP A, SEXP B)        \
+    EXTERNC afr_##fn##_na(SEXP A, SEXP B)        \
     {                                           \
         try{                                    \
             double a = *RealPtr(A, 0);          \

--- a/src/classify.cpp
+++ b/src/classify.cpp
@@ -41,7 +41,7 @@ array knn1(array &train_feats, array &test_feats, array &train_labels)
     return train_labels(idx.as(f32)).T();
 }
 
-EXTERNC af_knn1(SEXP trainFeats, SEXP testFeats, SEXP trainLabels)
+EXTERNC afr_knn1(SEXP trainFeats, SEXP testFeats, SEXP trainLabels)
 {
     try {
         array *train_feats  = getPtr(trainFeats);

--- a/src/core.cpp
+++ b/src/core.cpp
@@ -3,7 +3,7 @@
 using namespace af;
 
 #define REDUCE(fn)                              \
-    EXTERNC af_##fn(SEXP A, SEXP _dim)          \
+    EXTERNC afr_##fn(SEXP A, SEXP _dim)          \
     {                                           \
         try {                                   \
             array *a = getPtr(A);               \
@@ -20,7 +20,6 @@ using namespace af;
     }                                           \
 
 REDUCE(sum)
-REDUCE(mul)
 REDUCE(anytrue)
 REDUCE(alltrue)
 REDUCE(max)
@@ -30,21 +29,20 @@ REDUCE(median)
 
 #undef REDUCE
 
-EXTERNC af_accum(SEXP A, SEXP _dim, SEXP _OP)
+EXTERNC afr_accum(SEXP A, SEXP _dim)
 {
     try {
         array *a = getPtr(A);
         int dim = (int)*RealPtr(_dim, 0);
-        int op  = (int)*RealPtr(_OP, 0);
         array *b = new array();
-        *b = accum(*a, dim, (af_op_t)op);
+        *b = accum(*a, dim);
         return getSEXP(b);
     } catch (af::exception &ae) {
         error_return(ae.what());
     }
 }
 
-EXTERNC af_where(SEXP A)
+EXTERNC afr_where(SEXP A)
 {
     try {
         array *a = getPtr(A);
@@ -56,7 +54,7 @@ EXTERNC af_where(SEXP A)
     }
 }
 
-EXTERNC af_count(SEXP A)
+EXTERNC afr_count(SEXP A)
 {
     try {
         array *a = getPtr(A);

--- a/src/eval.cpp
+++ b/src/eval.cpp
@@ -6,7 +6,7 @@
 using namespace af;
 af::array *getPtr(SEXP S);
 
-extern "C" SEXP af_eval_r(SEXP A)
+extern "C" SEXP afr_eval(SEXP A)
 {
     try {
         array *a = getPtr(A);

--- a/src/index.cpp
+++ b/src/index.cpp
@@ -2,14 +2,7 @@
 #include "R_wrapper.h"
 using namespace af;
 
-
-/// Specify which address-space pointer belongs
-typedef enum {
-    afDevice, ///< Device-memory pointer
-    afHost    ///< Host-memory pointer
-} af_source_t;
-
-EXTERNC af_get_nnss(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
+EXTERNC afr_get_nnss(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
 {
     // i is numeric array
     // j is numeric array
@@ -36,7 +29,7 @@ EXTERNC af_get_nnss(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
     return getSEXP(res);
 }
 
-EXTERNC af_get_aass(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
+EXTERNC afr_get_aass(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
 {
     // i is arrayfire array
     // j is arrayfire array
@@ -49,7 +42,7 @@ EXTERNC af_get_aass(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
     return getSEXP(res);
 }
 
-EXTERNC af_get_anss(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
+EXTERNC afr_get_anss(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
 {
     // i is arrayfire array
     // j is numeric array
@@ -69,7 +62,7 @@ EXTERNC af_get_anss(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
     return getSEXP(res);
 }
 
-EXTERNC af_get_nass(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
+EXTERNC afr_get_nass(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
 {
     // i is numeric array
     // j is arrayfire array
@@ -89,7 +82,7 @@ EXTERNC af_get_nass(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
     return getSEXP(res);
 }
 
-EXTERNC af_get_sass(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
+EXTERNC afr_get_sass(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
 {
     // j is arrayfire array
     // i, k,l are spans and can be ignored
@@ -100,7 +93,7 @@ EXTERNC af_get_sass(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
     return getSEXP(res);
 }
 
-EXTERNC af_get_asss(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
+EXTERNC afr_get_asss(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
 {
     // i is arrayfire array
     // j, k,l are spans and can be ignored
@@ -112,7 +105,7 @@ EXTERNC af_get_asss(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
 }
 
 
-EXTERNC af_get_snss(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
+EXTERNC afr_get_snss(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
 {
     // j is numeric array
     // i, k,l are spans and can be ignored
@@ -129,7 +122,7 @@ EXTERNC af_get_snss(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
     return getSEXP(res);
 }
 
-EXTERNC af_get_nsss(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
+EXTERNC afr_get_nsss(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
 {
     // i is numeric array
     // j, k,l are spans and can be ignored
@@ -146,7 +139,7 @@ EXTERNC af_get_nsss(SEXP x, SEXP i, SEXP j, SEXP k, SEXP l)
     return getSEXP(res);
 }
 
-EXTERNC af_set_nnss(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
+EXTERNC afr_set_nnss(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
 {
     // i is numeric array
     // j is numeric array
@@ -174,7 +167,7 @@ EXTERNC af_set_nnss(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
     return getSEXP(res);
 }
 
-EXTERNC af_set_aass(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
+EXTERNC afr_set_aass(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
 {
     // i is arrayfire array
     // j is arrayfire array
@@ -188,7 +181,7 @@ EXTERNC af_set_aass(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
     return getSEXP(res);
 }
 
-EXTERNC af_set_anss(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
+EXTERNC afr_set_anss(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
 {
     // i is arrayfire array
     // j is numeric array
@@ -207,7 +200,7 @@ EXTERNC af_set_anss(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
     return getSEXP(res);
 }
 
-EXTERNC af_set_nass(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
+EXTERNC afr_set_nass(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
 {
     // i is numeric array
     // j is arrayfire array
@@ -227,7 +220,7 @@ EXTERNC af_set_nass(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
     return getSEXP(res);
 }
 
-EXTERNC af_set_sass(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
+EXTERNC afr_set_sass(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
 {
     // j is arrayfire array
     // i, k,l are spans and can be ignored
@@ -239,7 +232,7 @@ EXTERNC af_set_sass(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
     return getSEXP(res);
 }
 
-EXTERNC af_set_asss(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
+EXTERNC afr_set_asss(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
 {
     // i is arrayfire array
     // j, k,l are spans and can be ignored
@@ -252,7 +245,7 @@ EXTERNC af_set_asss(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
 }
 
 
-EXTERNC af_set_snss(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
+EXTERNC afr_set_snss(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
 {
     // j is numeric array
     // i, k,l are spans and can be ignored
@@ -269,7 +262,7 @@ EXTERNC af_set_snss(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
     return getSEXP(res);
 }
 
-EXTERNC af_set_nsss(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
+EXTERNC afr_set_nsss(SEXP x, SEXP v, SEXP i, SEXP j, SEXP k, SEXP l)
 {
     // i is numeric array
     // j, k,l are spans and can be ignored

--- a/src/stats.cpp
+++ b/src/stats.cpp
@@ -3,7 +3,7 @@
 using namespace af;
 
 #define REDUCE(fn)                              \
-    EXTERNC af_##fn(SEXP A)                     \
+    EXTERNC afr_##fn(SEXP A)                     \
     {                                           \
         try {                                   \
             array *a = getPtr(A);               \
@@ -16,12 +16,11 @@ using namespace af;
     }                                           \
 
 REDUCE(stdev)
-REDUCE(cov)
 REDUCE(var)
 #undef REDUCE
 
 #define REDUCE(fn)                              \
-    EXTERNC af_##fn##2(SEXP A, SEXP B)          \
+    EXTERNC afr_##fn##2(SEXP A, SEXP B)         \
     {                                           \
         try {                                   \
             array *a = getPtr(A);               \
@@ -35,4 +34,3 @@ REDUCE(var)
     }                                           \
 
 REDUCE(cov)
-REDUCE(var)

--- a/src/unary.cpp
+++ b/src/unary.cpp
@@ -3,11 +3,11 @@
 using namespace af;
 
 #define UNARY(fn)                               \
-    EXTERNC af_##fn(SEXP A)                     \
+    EXTERNC afr_##fn(SEXP A)                    \
     {                                           \
         try {                                   \
-            array *a = getPtr(A);               \
-            array *b = new array();             \
+            af::array *a = getPtr(A);           \
+            af::array *b = new array();         \
             *b = fn(*a);                        \
             return getSEXP(b);                  \
         } catch (af::exception &ae) {           \
@@ -30,7 +30,7 @@ UNARY(acosh)
 UNARY(atanh)
 
 UNARY(log)
-UNARY(log2)
+//UNARY(log2)
 UNARY(log10)
 UNARY(log1p)
 
@@ -38,9 +38,9 @@ UNARY(exp)
 UNARY(expm1)
 UNARY(erf)
 UNARY(erfc)
-UNARY(erfinv)
-UNARY(gamma)
-UNARY(gammaln)
+//UNARY(erfinv)
+UNARY(tgamma)
+UNARY(lgamma)
 
 UNARY(abs)
 UNARY(sqrt)
@@ -48,8 +48,8 @@ UNARY(sqrt)
 UNARY(round)
 UNARY(ceil)
 UNARY(floor)
-UNARY(trunc)
-UNARY(sign)
+//UNARY(trunc)
+//UNARY(sign)
 
 UNARY(real)
 UNARY(imag)

